### PR TITLE
fix(vllm): initialize prompt logprobs before prefix cache policy

### PR DESCRIPTION
**Pull request by git-jxj**
## Summary

When a token request asks for `prompt_logprobs`, Dynamo currently assigns it after constructing vLLM `SamplingParams`. The constructor has already derived `skip_reading_prefix_cache=False`, so a repeated prompt incorrectly reuses cached tokens that have no prompt logits.

Pass the requested `prompt_logprobs` into the constructor so vLLM sets its derived cache policy. Preserve model defaults, RL token-in behavior, and explicit cache-policy overrides.

## Validation

- CPU reproduction with the actual vLLM `KVCacheManager`: after warming a 39-token prompt, the original Dynamo parameters reuse 32 tokens for `prompt_logprobs=0`; native vLLM and the fix reuse zero.
- Final regression cases on the original function: 2 failed, 3 passed. Covers omitted/zero prompt logprobs, ordinary/RL requests, and explicit cache policy. Nonempty model defaults distinguish ordinary temperature inheritance from RL defaults.
- `python -m pytest components/src/dynamo/vllm/tests/test_vllm_tito_parity.py components/src/dynamo/vllm/tests/test_vllm_unit.py -q`: 191 passed.
- `pre-commit run --files` for the two changed files: passed.
- Real GPU HTTP comparison using Qwen3-0.6B BF16, pinned vLLM 0.28.0, RTX 5060 Ti 16GB, TP=1, eager/Triton attention, 16-token blocks, and prefix caching enabled. A fixed 265-token text prompt first demonstrates a normal 256-token cache hit.
- Original code: repeated requests with `prompt_logprobs=0` and `5` still report 256 cached tokens and omit `nvext.prompt_logprobs`, despite HTTP 200 and normal generated completions.
- Fixed code: the same requests report zero cached tokens and return all 265 prompt-logprob positions (first entry null; every remaining position contains the actual prompt token and finite probabilities). Width 0 returns one entry; width 5 returns 5–6 entries including the actual token. Ordinary repeated requests still hit 256 cached tokens. All four real HTTP cases pass.

## Related Issues

- [x] Confirmed — no related issue. Found while auditing the current main branch.

### CODE REVIEW

File: components/src/dynamo/vllm/tests/test_vllm_tito_parity.py [hunk 1]
```diff
@@ -377,6 +377,90 @@ def test_engine_data_preserves_prompt_logprobs(self):
         assert final["engine_data"]["completion_token_ids"] == [11]
 
 
+class TestPromptLogprobsPrefixCaching:
+    @pytest.mark.parametrize("prompt_logprobs", [None, 0, 5])
+    @pytest.mark.parametrize("enable_rl", [False, True])
+    def test_prompt_logprobs_recompute_cached_prefix(
+        self, prompt_logprobs, enable_rl, monkeypatch
+    ):
```
[the remark this answers is not in the document]
Comment (git-jxj) [reply components/src/dynamo/vllm/tests/test_vllm_tito_parity.py:385]: These tests inherit the module-level pytestmark declared near the top of this file: unit, pre_merge, vllm, core, and gpu_0. I verified the effective markers using actual pytest collection; all new parametrized cases and the explicit-override test inherit the required markers. The pytest-marker-report pre-commit gate also passes, so additional function-level marker duplication is unnecessary.

File: components/src/dynamo/vllm/tests/test_vllm_tito_parity.py [hunk 2]
```diff
@@ -377,6 +377,90 @@ def test_engine_data_preserves_prompt_logprobs(self):
         assert final["engine_data"]["completion_token_ids"] == [11]
 
 
+class TestPromptLogprobsPrefixCaching:
+    @pytest.mark.parametrize("prompt_logprobs", [None, 0, 5])
```
[the remark this answers is not in the document]
Comment (git-jxj) [reply components/src/dynamo/vllm/tests/test_vllm_tito_parity.py]: Refined in dded1548ef: removed the redundant positive-width cache scenarios and kept None/0. Both ordinary and RL paths remain covered with nonempty model defaults: ordinary requests inherit temperature=0.7 while RL requests use 1.0, so the two branches now verify distinct behavior. The original function fails both zero-logprob cache cases; the fixed class passes all five tests and the two related test files pass all 191 cases. Separate real GPU HTTP validation also covers widths 0 and 5.

**Comment by git-jxj** [comment]
[Pre Merge](https://github.com/ai-dynamo/dynamo/actions/runs/34320811560) passed for current head `d7f69a568`. Full CI has not run for this commit, and the upstream `pull-request/14323` branch has not been created, so the required `backend-status-check`, `dynamo-status-check`, and `deploy-status-check` results are still unreported.

Could a maintainer authorize this revision with `/ok to test d7f69a568`? Code-owner review is also still outstanding.

**Comment by git-jxj** [comment]
Following up for current head `421db2181` after the branch update. Pre Merge passed for this revision, but the earlier CI authorization request named `d7f69a568`, and full CI results are still unreported for `421db2181`. Could a maintainer authorize this head with `/ok to test 421db2181` and review the PR when available?
